lib/Optimizer/Transforms/GenDeviceCodeLoader.cpp

```diff
@@ -10,16 +10,43 @@
 #include "cudaq/Optimizer/Builder/Factory.h"
 #include "cudaq/Optimizer/Builder/Runtime.h"
 #include "cudaq/Optimizer/Transforms/Passes.h"
+#include "llvm/ADT/DepthFirstIterator.h"
 #include "llvm/Support/Debug.h"
 #include "llvm/Support/FileSystem.h"
 #include "llvm/Support/ToolOutputFile.h"
+#include "mlir/Analysis/CallGraph.h"
 #include "mlir/IR/Diagnostics.h"
 #include "mlir/Transforms/Passes.h"
 
 #define DEBUG_TYPE "device-code-loader"
 
 using namespace mlir;
 
+namespace llvm {
+// FIXME: `GraphTraits<const mlir::CallGraphNode *>` as defined in
```

Comment:
Not sure I understand this. Did you look at `Optimizer/CodeGen/TranslateToOpenQASM.cpp`? It is using `mlir::CallGraph`.